### PR TITLE
Path: listdir method to return directory entry names

### PR DESCRIPTION
**Pull request by stdweird**
Based on #243

### CODE REVIEW

**Comment by stdweird** [comment]
Main reason to wrap this is to able to unittest readdir in maven-tools with the mocked files

File: src/main/perl/FileWriter.pm [hunk 1]
```diff
@@ -301,10 +303,21 @@ sub close
                 $msg = 'would have been';
                 $self->debug(1, "File $filename with NoAction=1");
             } else {
+                my $parent_dir = dirname($filename);
+                # Pass NoAction here, as it keeps track of the NoAction value during initialisation and/or keeps_state
+                my $cafpath = CAF::Path::mkcafpath(log => *$self->{LOG}, NoAction => $options->{noaction});
+                # only create the directory if it didn't exist yet
+                # if not, this would for the directory mode on existing directories
```
Comment (ned21) [inline src/main/perl/FileWriter.pm]: This line doesn't make sense. "this would [what?] for the"

**Comment by stdweird** [comment]
@ned21 i'll address the remark, but i made a mistake in the description. this is based on the already merged PR #243; and your remark is from that PR. i'll add a new commit with better description

**Comment by ned21** [comment]
@stdweird I thought I had seen that odd sentence before, and assumed I had just reviewed this PR before!

**Comment by stdweird** [comment]
@ned21 i added a new commit to this PR

**Comment by stdweird** [comment]
retest this please

**Pull request merged**